Does not build with upcoming Go 1.22

### What's wrong?

When attempting to build with Go 1.22rc1 it fails to build.

Noticed with  v0.38.1 in 
* https://github.com/Homebrew/homebrew-core/pull/157782

### Steps to reproduce

1. Install Go 1.22rc1
2. Run `make binaries`
3. See build error:

````
# github.com/grafana/pyroscope-go/godeltaprof/internal/pprof
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/grafana/pyroscope-go/godeltaprof@v0.1.3/internal/pprof/delta_mutex.go:25:20: undefined: runtime_cyclesPerSecond
/Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/grafana/pyroscope-go/godeltaprof@v0.1.3/internal/pprof/proto.go:400:8: undefined: runtime_expandFinalInlineFrame
````

### System information

All, irrelevant

### Software version

Grafana Agent v0.39.0

### Configuration

```text
No special config, just building using default `Makefile`.
```


### Logs

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -43,6 +43,8 @@ Main (unreleased)
 
 - Split instance ID and component groupings into separate panels for `remote write active series by component` in the Flow mixin. (@tristanburgess)
 
+- Updated dependency to add support for Go 1.22 (@stefanb)
+
 v0.39.0 (2024-01-09)
 --------------------
 

--- a/go.mod
+++ b/go.mod
@@ -57,7 +57,7 @@ require (
 	github.com/grafana/go-gelf/v2 v2.0.1
 	// Loki main commit where the Prometheus dependency matches ours. TODO(@tpaschalis) Update to kXYZ branch once it's available
 	github.com/grafana/loki v1.6.2-0.20231004111112-07cbef92268a
-	github.com/grafana/pyroscope-go/godeltaprof v0.1.3
+	github.com/grafana/pyroscope-go/godeltaprof v0.1.6
 	github.com/grafana/pyroscope/api v0.2.0
 	github.com/grafana/pyroscope/ebpf v0.4.1
 	github.com/grafana/regexp v0.0.0-20221123153739-15dc172cd2db

